### PR TITLE
Add missing `.editorconfig` settings present in dotnet/runtime

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

Add settings from https://github.com/dotnet/runtime/blob/master/.editorconfig

We may want a different value for of the new settings, so please leave review comments,

## PR Context

#7357

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -99,17 +102,38 @@ dotnet_style_collection_initializer = true:suggestion
 dotnet_style_coalesce_expression = true:suggestion
 dotnet_style_null_propagation = true:suggestion
 dotnet_style_explicit_tuple_names = true:suggestion
+dotnet_style_readonly_field = true:suggestion
+dotnet_style_prefer_is_null_check_over_reference_equality_method = true:suggestion
```
Comment (iSazonov) [inline .editorconfig:106]: With the setting we should replace all null reference checks with "is null" to fix Roslyn analyzer issues.

File: .editorconfig [hunk 2]
```diff
@@ -48,6 +48,8 @@ indent_size = 2
 # Sort using and Import directives with System.* appearing first
 dotnet_sort_system_directives_first = true
 
+csharp_using_directive_placement = outside_namespace:suggestion
```
Comment (iSazonov) [inline .editorconfig:51]: Please move to csharp section below.

File: .editorconfig [hunk 3]
```diff
@@ -99,17 +102,38 @@ dotnet_style_collection_initializer = true:suggestion
...
+dotnet_style_prefer_inferred_tuple_names = true:suggestion
+dotnet_style_prefer_inferred_anonymous_type_member_names = true:suggestion
+dotnet_style_prefer_conditional_expression_over_return = true:refactoring
+dotnet_style_prefer_conditional_expression_over_assignment = true:refactoring
+dotnet_style_prefer_auto_properties = true:suggestion
+csharp_prefer_simple_default_expression = true:suggestion
+
+dotnet_code_quality_unused_parameters = non_public:suggestion
+dotnet_code_quality.ca1802.api_surface = private, internal
+dotnet_code_quality.CA2208.api_surface = public
```
Comment (iSazonov) [inline .editorconfig:116]: Please add comments about that it is.

File: .editorconfig [hunk 4]
```diff
@@ -99,17 +102,38 @@ dotnet_style_collection_initializer = true:suggestion
...
 
 # CSharp code style settings:
 [*.cs]
 
 csharp_preserve_single_line_blocks = true
 csharp_preserve_single_line_statements = false
+csharp_prefer_braces = true:refactoring
```
Comment (iSazonov) [inline .editorconfig:123]: What is refactoring here?
Can you point a description?
I'd expect `true`.
https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-language-conventions?view=vs-2019#code-block-preferences

File: .editorconfig [hunk 5]
```diff
@@ -118,6 +142,9 @@ csharp_style_var_for_built_in_types = false:none
 csharp_style_var_when_type_is_apparent = false:none
 csharp_style_var_elsewhere = false:suggestion
 
+# Expression-bodied members
+csharp_style_expression_bodied_local_functions = true:refactoring
```
Comment (iSazonov) [inline .editorconfig:146]: The same.

File: .editorconfig (hunk 1 above)
Comment (vexx32) [reply .editorconfig:106]: Yeah, that'd be best. Would probably be best to put that in its own PR, there's going to be quite a lot of those I reckon. 😁

Comment (iSazonov) [reply .editorconfig:106]: We can accept this. Fixing code should be in follow PR.

File: .editorconfig (hunk 3 above)
Comment (xtqqczze) [reply .editorconfig:116]: I will consider `dotnet_code_quality.ca1802` and `dotnet_code_quality.CA2208` seperately from this PR

File: .editorconfig (hunk 4 above)
Comment (xtqqczze) [reply .editorconfig:123]: It seems `refactoring` is not valid in this context. In other contexts it is a symonym for `silent`
https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-naming-conventions?view=vs-2019#severity

Comment (xtqqczze) [reply .editorconfig]: issue opened upstream: https://github.com/dotnet/runtime/pull/37367

**Comment by xtqqczze** [comment]
Opened #12886 for `PowerShell-CI-static-analysis` failure.

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@xtqqczze Please rebase to fix CI-static.

**Comment by TravisEz13** [comment]
cc @daxian-dbw @adityapatwardhan  Can you review?

**Comment by TravisEz13** [comment]
cc @daxian-dbw @adityapatwardhan Can you review?

**Review by adityapatwardhan — APPROVED** [review]

**Comment by xtqqczze** [comment]
@daxian-dbw Can you please review?

**Pull request merged**